src/sync/app_data.py

```diff
@@ -1,154 +1,27 @@
-                    f"No content found after {attempts} attempts for {app_hash} assuming NULL."
-                )
-                self._found.append(
-                    FoundContent(
-                        app_hash=app_hash,
-                        first_seen_block=row.first_seen_block,
-                        content={},
-                    )
-                )
-            else:
-                self._not_found.append(row)
-
-    def write_found_content(self) -> None:
-        assert len(self.new_rows) == 0, "Must call _handle_new_records first!"
-        self.file_manager.write_ndjson(
-            data=[x.as_dune_record() for x in self._found], name=self.content_filename
-        )
-        # When not_found is empty, we want to overwrite the file (hence skip_empty=False)
-        # This happens when number of attempts exceeds GIVE_UP_THRESHOLD
-        self.file_manager.write_ndjson(
-            data=[x.as_dune_record() for x in self._not_found],
-            name=self.missing_file_name,
-            skip_empty=False,
-        )
-
-    def write_sync_data(self) -> None:
-        # Only write these if upload was successful.
-        self.file_manager.write_csv(
-            data=[{self.config.sync_column: str(self.block_range.block_to)}],
-            name=self.config.sync_file,
-        )
-
-    async def fetch_content_and_filter(
-        self, max_retries: int, give_up_threshold: int
-    ) -> None:
-        """
-        Run loop fetching app_data for hashes,
-        separates into (found and not found), returning the pair.
-        """
-        await self._handle_new_records(max_retries)
-        log.info(
-            f"Attempting to recover missing {len(self.missing_values)} records from previous run"
-        )
-        await self._handle_missing_records(max_retries, give_up_threshold)
-
-
 async def sync_app_data(
-    aws: AWSClient,
-    dune: DuneFetcher,
+    orderbook: OrderbookFetcher,
+    dune: DuneClient,
     config: AppDataSyncConfig,
-    ipfs_access_key: str,
     dry_run: bool,
 ) -> None:
     """App Data Sync Logic"""
-    block_range = BlockRange(
-        block_from=last_sync_block(
-            aws,
-            table=SYNC_TABLE,
-            genesis_block=12153262,  # First App Hash Block
-        ),
-        block_to=await dune.latest_app_hash_block(),
-    )
-
-    data_handler = AppDataHandler(
-        file_manager=FileIO(config.volume_path / str(SYNC_TABLE)),
-        new_rows=await dune.get_app_hashes(block_range),
-        block_range=block_range,
-        config=config,
-        ipfs_access_key=ipfs_access_key,
-        missing_file_name=config.missing_files_name,
-    )
-    await data_handler.fetch_content_and_filter(
-        max_retries=config.max_retries, give_up_threshold=config.give_up_threshold
-    )
-    UploadHandler(aws, data_handler, table=SYNC_TABLE).write_and_upload_content(dry_run)
+    hashes = orderbook.get_app_hashes()
+    if not dry_run:
+        dune.upload_csv(
```

Comment:
Just to double-check, this essentially kills the old table and creates a new one with the updated csv, right?

Reply:
Yes, it overrides any table with the same name